internal/catalogmetadata/client/client.go

```diff
@@ -31,14 +75,75 @@ func (c *Client) Bundles(ctx context.Context) ([]*catalogmetadata.Bundle, error)
 		return nil, err
 	}
 	for _, catalog := range catalogList.Items {
-		channels, err := fetchCatalogMetadata[catalogmetadata.Channel](ctx, c.cl, catalog.Name, declcfg.SchemaChannel)
+		channels := []*catalogmetadata.Channel{}
+		bundles := []*catalogmetadata.Bundle{}
+
+		// TODO: When a catalogd release containing https://github.com/operator-framework/catalogd/pull/168
+		// is made this should be updated in favor of parsing `Catalog.Status` for the
+		// appropriate URL for fetching catalog contents.
+		req, err := http.NewRequestWithContext(ctx, http.MethodGet, fmt.Sprintf(catalogdOnClusterURLTemplate, c.baseURL, catalog.Name), nil)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("error forming request: %s", err)
 		}
 
-		bundles, err := fetchCatalogMetadata[catalogmetadata.Bundle](ctx, c.cl, catalog.Name, declcfg.SchemaBundle)
+		cacheData, ok := c.cache[catalog.Name]
```

Comment:
I imagine your `Client` is used concurrently? You likely need a `sync.RWMutex` around access to this member?

Reply:
Should be taken care of in the new cache implementation as of https://github.com/operator-framework/operator-controller/pull/411/commits/fe924c2c2f9c686d3720bb930d42f01590409df1